app/controllers/BusinessContactDetailsController.scala

```diff
@@ -40,29 +42,40 @@ class BusinessContactDetailsController @Inject() (
   getData: DataRetrievalActionProvider,
   requireData: DataRequiredAction,
   formProvider: BusinessContactDetailsFormProvider,
+  userRoleProvider: UserRoleProvider,
   val controllerComponents: MessagesControllerComponents,
-  view: BusinessContactDetailsView
+  view: BusinessContactDetailsView,
+  appConfig: FrontendAppConfig
 )(implicit ec: ExecutionContext)
     extends FrontendBaseController
     with I18nSupport {
 
-  val form = formProvider()
-
   def onPageLoad(mode: Mode, draftId: DraftId): Action[AnyContent] =
     (identify andThen getData(draftId) andThen requireData) {
       implicit request =>
-        val preparedForm = BusinessContactDetailsPage.fill(form)
+        val incCompanyName =
+          userRoleProvider
+            .getUserRole(request.userAnswers)
+            .contactDetailsIncludeCompanyName && appConfig.agentOnBehalfOfTrader
+        val form           = formProvider(incCompanyName)
```

Comment:
this block could be refactored into a def since it is copied for the onSubmit method

Reply:
I hadn't because the form and the view both need to know the incCompanyName value.